File: tests/transpiler/x/java/record_assign.java
public class Main {
    static Counter c = new Counter(0);

    static void inc(Counter c) {
c.put("n", ((Integer) (c.get("n"))) + 1);
    }
    public static void main(String[] args) {
        inc(c);
        System.out.println(((Integer) (c.get("n"))));
    }
}

File: archived/tests/transpiler/x/pas/record_assign.pas
{$mode objfpc}
program Main;
type Counter = record
  n: integer;
end;
procedure inc(c: Counter);
begin
  c.n := c.n + 1;
end;
var
  c: Counter;
begin
  c.n := 0;
  inc(c);
  writeln(c.n);
end.
